components/site/src/link_checking.rs

```diff
@@ -101,53 +104,94 @@ pub fn check_external_links(site: &Site) -> Result<()> {
         .values()
         .map(|p| {
             let path = &p.file.path;
-            p.external_links.iter().map(move |l| (path.clone(), l))
+            p.external_links.iter().map(move |l| {
+                let domain = match Url::parse(l) {
+                    Ok(url) => match url.host_str().map(String::from) {
+                        Some(domain_str) => domain_str,
+                        None => "".to_string(),
+                    },
+                    Err(why) => panic!("{:?}", why),
```

Comment:
Not sure what to do here. I'm pretty sure we would never get an error since the links are already parsed by the markdown renderer?

Reply:
They are not parsed afaik so they could be invalid.
I think it's a case where it would probably be easier to follow the code if we add errors by using a for loop rather than iterators.

Reply:
Done in https://github.com/getzola/zola/pull/1421/commits/72f21d7b578422abfa84ed6b1e912bb03fc2f430, it looks simpler to me. What do you think? Also, is the error handling ok?